### PR TITLE
RM-2641: Update rules for ABC cop

### PR DESCRIPTION
**Pull request by wescreations**
Helps reduce the number of ABC violations. 

ABC metric assumes we're making additional method calls when we doing something like

school.name = name

disabling this rule on this cop just reduces violations by lowering the ABC score in certain scenarios. [You can read about it here](https://stackoverflow.com/questions/71984594/rubocop-complains-about-metrics-abcsize)

You'll see the one actual change, there was 6 other files that magically appeared while updating them. Anyone with context just let me know if I need to delete or keep those.

This branch is relied upon by [this PR here](https://github.com/RaiseMe/raise-ror-webapp/pull/14981)

Basically, you need to switch to this branch on ruby-commons, then you can switch back to olympus and run rubocop lib app over there.

### CODE REVIEW

**Review by Trevor-Warthman-2 — APPROVED** [review]
Didn't test but I trust you completely with my life
Can leave in or remoge the idea files those will just mess with intellij

**Review by toddnestor — APPROVED** [review]
